src/documentProviders/index.ts

```diff
@@ -14,18 +14,31 @@ export abstract class ResourceDocumentProvider implements vscode.TextDocumentCon
   abstract provideTextDocumentContent(uri: vscode.Uri): Promise<string>;
 
   /**
-   * Convert a resource object to a URI that can be used to display the resource in a read-only editor.
-   * @param resource The resource object
-   * @param filename The filename to use for the URI
+   * Base implementation of converting a resource object to a URI that can be used to display the resource in a read-only editor.
```

Comment:
Migrate the main logic for producing the URI for a document provider + resource into a static method, so that callpoints in commands won't need to make a bogus document provider instance just for the purposes of making such a URI.